src/Modules/Scoreboard/Services/ScoreboardService.cs

```diff
@@ -0,0 +1,144 @@
+            "Scoreboard.Scoreboard",
+            await GetData()
+        );
+        await SendRoundsInfo(playerLogin);
+    }
+
+    public async Task ShowScoreboard()
+    {
+        await _manialinks.SendManialinkAsync("Scoreboard.Scoreboard", await GetData());
+    }
+
+    private async Task<dynamic> GetData()
+    {
+        return new
+        {
+            MaxPlayers = await GetMaxPlayersAsync(),
+            RoundsPerMap = _roundsPerMap,
+            PositionColors = new Dictionary<int, string> { { 1, "d1b104" }, { 2, "9e9e9e" }, { 3, "915d29" } },
+            headerColor = _config.Theme.UI.HeaderBackgroundColor,
+            primaryColor = _config.Theme.UI.PrimaryColor,
+            positionBackgroundColor = _config.Theme.UI.Scoreboard.PositionBackgroundColor,
+            backgroundColor = _config.Theme.UI.BackgroundColor,
+            playerRowHighlightColor = _config.Theme.UI.Scoreboard.PlayerRowHighlightColor,
+            logoUrl = _config.Theme.UI.LogoUrl
+        };
+    }
+
+    public async Task HideNadeoScoreboard()
+    {
+        var hudSettings = new List<string>()
+        {
+            @"{
+    ""uimodules"": [
+        {
+            ""id"": ""Race_ScoresTable"",
+            ""position"": [-50,0],
+            ""scale"": 1,
+            ""visible"": false,
+            ""visible_update"": true
+        }
+    ]
+}"
+        };
+
+        await _server.Remote.TriggerModeScriptEventArrayAsync("Common.UIModules.SetProperties", hudSettings.ToArray());
+    }
+
+    public async Task ShowNadeoScoreboard()
+    {
+        var hudSettings = new List<string>()
+        {
+            @"{
+    ""uimodules"": [
+        {
+            ""id"": ""Race_ScoresTable"",
+            ""position"": [-50,0],
+            ""scale"": 1,
+            ""visible"": true,
+            ""visible_update"": true
+        }
+    ]
+}"
+        };
+
+        await _server.Remote.TriggerModeScriptEventArrayAsync("Common.UIModules.SetProperties", hudSettings.ToArray());
+    }
+
+    public async Task SendRoundsInfo()
+    {
+        await _manialinks.SendManialinkAsync("Scoreboard.RoundsInfo",
+            new { RoundsPerMap = _roundsPerMap, CurrentRound = _currentRound });
+    }
+
+    public async Task SendRoundsInfo(string playerLogin)
+    {
+        await _manialinks.SendManialinkAsync(
+            await _playerManager.GetPlayerAsync(PlayerUtils.ConvertLoginToAccountId(playerLogin)),
+            "Scoreboard.RoundsInfo",
+            new { RoundsPerMap = _roundsPerMap, CurrentRound = _currentRound }
+        );
+    }
+
+    public async void LoadAndUpdateRoundsPerMap()
+    {
+        _roundsPerMap = await GetRoundsPerMapAsync();
+        await SendRoundsInfo();
+    }
+
+    public void SetCurrentRound(int round)
+    {
+        _currentRound = round;
+    }
+
+    private async Task<int> GetRoundsPerMapAsync()
+    {
+        var variables = await _server.Remote.GetModeScriptSettingsAsync();
```

Comment:
There is a `GetCurrentScriptSettingsAsync` within the `IMatchSettingsService` btw.

Reply:
ty, changed it in the new branch